src/app/library/toroidalMixer.js

```diff
@@ -0,0 +1,249 @@
+import Template from "./template";
+import paper from "paper";
+import ComponentPort from "../core/componentPort";
+import { Path, Point } from "paper/dist/paper-core";
```

Comment:
Make this Typescript. We might not port all the classes right now but new ones should be ported